modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptNestjsClientCodegen.java

```diff
@@ -0,0 +1,603 @@
+        supportingFiles.add(new SupportingFile("variables.mustache", getIndexDirectory(), "variables.ts"));
+        supportingFiles.add(new SupportingFile("encoder.mustache", getIndexDirectory(), "encoder.ts"));
+        supportingFiles.add(new SupportingFile("gitignore", "", ".gitignore"));
+        supportingFiles.add(new SupportingFile("git_push.sh.mustache", "", "git_push.sh"));
+        supportingFiles.add(new SupportingFile("README.mustache", getIndexDirectory(), "README.md"));
+
+        // determine NG version
+        SemVer nestVersion;
+        if (additionalProperties.containsKey(NEST_VERSION)) {
+            nestVersion = new SemVer(additionalProperties.get(NEST_VERSION).toString());
+        } else {
+            nestVersion = new SemVer(this.nestVersion);
+            LOGGER.info("generating code for Nestjs {} ...", nestVersion);
+            LOGGER.info("  (you can select the nestjs version by setting the additionalProperty nestVersion)");
+        }
+
+        if (additionalProperties.containsKey(NPM_NAME)) {
+            addNpmPackageGeneration(nestVersion);
+        }
+
+        if (additionalProperties.containsKey(STRING_ENUMS)) {
+            setStringEnums(Boolean.valueOf(additionalProperties.get(STRING_ENUMS).toString()));
+            additionalProperties.put("stringEnums", getStringEnums());
+            if (getStringEnums()) {
+                enumSuffix = "";
+                classEnumSeparator = "";
+            }
+        }
+
+        if (additionalProperties.containsKey(WITH_INTERFACES)) {
+            boolean withInterfaces = Boolean.parseBoolean(additionalProperties.get(WITH_INTERFACES).toString());
+            if (withInterfaces) {
+                apiTemplateFiles.put("apiInterface.mustache", "Interface.ts");
+            }
+        }
+
+        if (additionalProperties.containsKey(TAGGED_UNIONS)) {
+            taggedUnions = Boolean.parseBoolean(additionalProperties.get(TAGGED_UNIONS).toString());
+        }
+
+        if (nestVersion.atLeast("6.0.0")) {
+            if (!additionalProperties.containsKey(PROVIDED_IN_ROOT)) {
+                additionalProperties.put(PROVIDED_IN_ROOT, true);
+            } else {
+                additionalProperties.put(PROVIDED_IN_ROOT, Boolean.parseBoolean(
+                    additionalProperties.get(PROVIDED_IN_ROOT).toString()
+                ));
+            }
+        } else {
+            additionalProperties.put(PROVIDED_IN_ROOT, false);
+        }
+
+        additionalProperties.put(NEST_VERSION, nestVersion);
+        additionalProperties.put("injectionToken", nestVersion.atLeast("4.0.0") ? "InjectionToken" : "OpaqueToken");
+        additionalProperties.put("injectionTokenTyped", nestVersion.atLeast("4.0.0"));
+        additionalProperties.put("useHttpClient", nestVersion.atLeast("4.3.0"));
+        additionalProperties.put("useRxJS6", nestVersion.atLeast("6.0.0"));        
```

Comment:
Looks like this logic was copied from the Angular generator. Does it apply to Nest framework with exactly same version numbers?

Reply:
No this line should be deleted